lib/buffer.js

```diff
@@ -1056,8 +1056,6 @@ Buffer.prototype.readDoubleBE = function readDoubleBE(offset, noAssert) {
 
 
 function checkInt(buffer, value, offset, ext, max, min) {
-  if (!isUint8Array(buffer))
```

Comment:
If `buffer` is no longer checked we can just pass `buffer.length` down. Not sure about the performance implication though.